Jint/Runtime/Interpreter/Expressions/JintBinaryExpression.cs

```diff
@@ -17,6 +30,85 @@ private JintBinaryExpression(Engine engine, BinaryExpression expression) : base(
             _right = Build(engine, expression.Right);
         }
 
+        protected abstract object EvaluateBinaryExpression();
+
+        protected override object EvaluateInternal()
```

Comment:
this part is a bit iffy for me, this is on performance hot path and now introduces another layer of indirection

Reply:
might require sealing, also minimize code size for non-default case, i.e. put if `if (!_engine.Options._IsOperatorOverloadingAllowed)` first and then call separate method for doing the operator overloading version?

Reply:
Removed the calls to virtual methods. Please check now.